enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -459,9 +529,13 @@ func UploadTree(ctx context.Context, env environment.Env, dirHelper *DirHelper,
 
 	// Upload output files to the remote cache and also add them to the local
 	// cache since they are likely to be used as inputs to subsequent actions.
-	if err := uploadFiles(ctx, uploader, env.GetFileCache(), filesToUpload); err != nil {
+	skippedFiles, skippedBytes, err := uploadMissingFiles(ctx, uploader, env, filesToUpload, instanceName, digestFunction)
+	if err != nil {
 		return nil, err
 	}
+	metrics.SkippedOutputBytes.Add(float64(skippedBytes))
+	txInfo.FileCount -= skippedFiles
+	txInfo.BytesTransferred -= skippedBytes
```

Comment:
nit: this subtraction after the fact feels slightly error prone, wdyt about passing txInfo as a param to uploadMissingFiles and having that func be responsible for incrementing FileCount/BytesTransferred? (would also let us avoid having 3 return values from that func)

Reply:
I'll do this in next change, since I have a few other related cleanups.